File: fixtures/java/java-multi-module/app/src/main/java/com/minos/fixture/multimodule/app/GreetingService.java
package com.minos.fixture.multimodule.app;

import com.minos.fixture.multimodule.api.GreetingPort;
import com.minos.fixture.multimodule.api.UserProfile;

import java.util.Objects;

public final class GreetingService {

    private final GreetingPort greetingPort;

    public GreetingService(GreetingPort greetingPort) {
        this.greetingPort = Objects.requireNonNull(greetingPort, "greetingPort");
    }

    public String greet(String name) {
        return greetingPort.greet(new UserProfile(name));
    }
}

File: fixtures/kotlin/kotlin-maven-simple/src/main/kotlin/com/minos/fixture/GreetingService.kt
package com.minos.fixture

interface GreetingPort { fun greet(name: String): String }

class GreetingService : GreetingPort {
    override fun greet(name: String): String = "Hello, $name"
}

class GreetingController(private val service: GreetingPort = GreetingService()) {
    fun message(name: String): String = service.greet(name)
}
